### PR TITLE
add quick starter sample for vertexai

### PR DESCRIPTION
**Pull request by cynthiajoan**
This starter sample are mostly identical to https://github.com/firebase/flutterfire/tree/master/packages/firebase_vertexai/firebase_vertexai/example

### CODE REVIEW

File: vertexai/lib/main.dart [hunk 1]
```diff
@@ -0,0 +1,548 @@
...
+  ) async =>
+      // This hypothetical API returns a JSON such as:
+      // {"base":"USD","date":"2024-04-17","rates":{"SEK": 0.091}}
+      {
+        'date': arguments['currencyDate'],
+        'base': arguments['currencyFrom'],
+        'rates': <String, Object?>{arguments['currencyTo']! as String: 0.091},
+      };
+
+  final exchangeRateTool = FunctionDeclaration(
```
Comment (morganchen12) [inline vertexai/lib/main.dart:114]: We may need to change this example. cc @rachelsaunders

**Review by morganchen12 — COMMENTED** [review]
LGTM minus the one comment.

File: vertexai/lib/main.dart (hunk 1 above)
Comment (cynthiajoan) [reply vertexai/lib/main.dart:114]: I double checked both generative function calling github sample as well as the documentation for it, seems it's not aligned. We will update this sample in the snippet project

Comment (rachelsaunders) [reply vertexai/lib/main.dart:114]: We'll soon be updating the snippets and overall "narrative" for the function calling page for both the Google AI SDKs and the Vertex AI for Firebase SDKs.  So, I wouldn't bother putting the _current_ snippets into GitHub just yet.

Comment (cynthiajoan) [reply vertexai/lib/main.dart:114]: sg, let me remove that part for now and the get the PR merged.

Comment (cynthiajoan) [reply vertexai/lib/main.dart]: done

**Review by morganchen12 — APPROVED** [review]

**Pull request merged**